src/tools/rbd_mirror/GroupReplayer.cc

```diff
@@ -1232,6 +1260,175 @@ void GroupReplayer<I>::update_mirror_group_replay_status() {
+      cls::rbd::MirrorImageSiteStatus mirror_image;
+      if (ir->is_running()) {
+        if (ir->is_replaying()) {
+          mirror_image.state = cls::rbd::MIRROR_IMAGE_STATUS_STATE_REPLAYING;
+        } else {
+          mirror_image.state = cls::rbd::MIRROR_IMAGE_STATUS_STATE_STARTING_REPLAY;
+        }
+      } else if (ir->is_stopped()) {
+        // FIXME: This does not capture the IR in error state
+        mirror_image.state = cls::rbd::MIRROR_IMAGE_STATUS_STATE_STOPPED;
+      } else {
+        mirror_image.state = cls::rbd::MIRROR_IMAGE_STATUS_STATE_STOPPING_REPLAY;
+      }
+      mirror_image.description = ir->get_state_description();
+
+      local_status.mirror_images[{ir->get_local_pool_id(),
+                                  ir->get_global_image_id()}] = mirror_image;
+      auto remote_pool_id = ir->get_remote_pool_id();
+      if (remote_pool_id >= 0) {
+        remote_status.mirror_images[{remote_pool_id,
+                                     ir->get_global_image_id()}] = mirror_image;
+      }
+    }
+  }
+// FIXME : Doesn't work at the moment
+  auto images_status = local_status.mirror_images;
+  // catch and bubble-up all the image level errors to group status
+  if (!images_status.empty() &&
+      ((m_local_group_ctx.primary && local_status.state == cls::rbd::MIRROR_GROUP_STATUS_STATE_STOPPED) ||
+      (!m_local_group_ctx.primary && local_status.state == cls::rbd::MIRROR_GROUP_STATUS_STATE_REPLAYING))) {
+    for (auto &image_site_status : images_status) {
+      if (image_site_status.second.state == cls::rbd::MIRROR_IMAGE_STATUS_STATE_ERROR) {
+        dout(10) << "ImageReplayer with global image id: " << image_site_status.first.global_image_id
+            << " in error state, with description: " << image_site_status.second.description
+            << " marking group replayer to error state with global group id: " << m_global_group_id
+            << dendl;
+        local_status.state = cls::rbd::MIRROR_GROUP_STATUS_STATE_ERROR;
+        local_status.description = "image in error state";
+        mirror_group_status_state = local_status.state;
+        break;
+      }
+    }
+  }
+  {
+    std::lock_guard locker{m_lock};
+    m_mirror_group_status_state = mirror_group_status_state;
+  }
+
+  // prevent the status from ping-ponging when failed replays are restarted
+  if (mirror_group_status_state &&
+      *mirror_group_status_state == cls::rbd::MIRROR_GROUP_STATUS_STATE_ERROR) {
+    local_status.state = *mirror_group_status_state;
+  }
+
+  m_local_status_updater->set_mirror_group_status(m_global_group_id,
+                                                  local_status, force,
+                                                  skip_image_status_update);
+  if (m_remote_group_peer.mirror_status_updater != nullptr) {
+    m_remote_group_peer.mirror_status_updater->set_mirror_group_status(
+        m_global_group_id, remote_status, force, skip_image_status_update);
+  }
+}
+
```

Comment:
Just wanted to confirm that you're just preserving this part of the code from previous  `set_mirror_group_status_update` code, right?

Reply:
Correct, except for the change where I pass `force` to the `set_mirror_group_status` instead of true/false.